### DISCUSSION TITLE
Use IPV4 only as egress?

### DISCUSSION THREAD

**Discussion by btrepp** [k3s-io/k3s#5620] [Q&A]
Hi All,

I'm trying to setup a mainly IPv6 setup, as I don't want complexity of two networks.
Unfortunately some external sites (eg github.com....) don't support IPv6, so I need some sort of IPv4 ability to access.

The problem is, as soon as any IPv4 is present, the entire cluster requires so much setup, I even need to now specify static ips on a DHCP network. It's honestly really easy to break the ipv4 side, when IPv6 works perfectly, but as the presence of Ipv4 means all the pods try to use that, and ignore the working ipv6 config :(.

I'd like to avoid all the complexity of IPV4, but have it work like 'nodes can get ipv4 addresses and use that to connect to github', use ipv6 where you can, but if you need an external ipv4 you can use this.

 Is this possible at all?. It seems annoying to have to meticuously manage an ipv4 network, just so one pod can access github.com to do a git clone :(.

I think there are some nat64 services, but now I have to run special dns servers. Just to work around a few sites that don't work on ipv6!.

Thanks

**Comment by brandond** [comment]
> The problem is, as soon as any IPv4 is present, the entire cluster requires so much setup, I even need to now specify static ips on a DHCP network.

Only the nodes themselves need static IPs, or at least a stable reservation within your DHCP server. The pod and cluster CIDRs themselves do not need to be routable across your LAN in order to use IPv4.

The way dual-stack and IPv6 are used at the moment, your nodes need an address in the IP family that the cluster uses. If you want to use IPv6; the nodes need IPv6 addresses, and an ipv6 default route. You might see if using the `--flannel-ipv6-masq` option (https://github.com/k3s-io/k3s/pull/4952) solves your problem, but I suspect it will not.
